tests/queries/0_stateless/01601_custom_tld.sql

```diff
@@ -1,16 +1,29 @@
-select 'no-tld';
-select cutToFirstSignificantSubdomainCustom('there-is-no-such-domain', 'public_suffix_list');
+select '-- no-tld';
 -- even if there is no TLD, 2-nd level by default anyway
 -- FIXME: make this behavior optional (so that TLD for host never changed, either empty or something real)
+select cutToFirstSignificantSubdomain('there-is-no-such-domain');
+select cutToFirstSignificantSubdomain('foo.there-is-no-such-domain');
+select cutToFirstSignificantSubdomain('bar.foo.there-is-no-such-domain');
+select cutToFirstSignificantSubdomainCustom('there-is-no-such-domain', 'public_suffix_list');
 select cutToFirstSignificantSubdomainCustom('foo.there-is-no-such-domain', 'public_suffix_list');
 select cutToFirstSignificantSubdomainCustom('bar.foo.there-is-no-such-domain', 'public_suffix_list');
 select firstSignificantSubdomainCustom('bar.foo.there-is-no-such-domain', 'public_suffix_list');
 
-select 'generic';
-select firstSignificantSubdomainCustom('foo.kernel.biz.ss', 'public_suffix_list'); -- kernel.biz.ss
+select '-- generic';
+select firstSignificantSubdomainCustom('foo.kernel.biz.ss', 'public_suffix_list'); -- kernel
 select cutToFirstSignificantSubdomainCustom('foo.kernel.biz.ss', 'public_suffix_list'); -- kernel.biz.ss
 
-select 'difference';
+select '-- difference';
 -- biz.ss is not in the default TLD list, hence:
 select cutToFirstSignificantSubdomain('foo.kernel.biz.ss'); -- biz.ss
 select cutToFirstSignificantSubdomainCustom('foo.kernel.biz.ss', 'public_suffix_list'); -- kernel.biz.ss
+
+select '-- 3+level';
+select cutToFirstSignificantSubdomainCustom('xx.blogspot.co.at', 'public_suffix_list'); -- xx.blogspot.co.at
+select firstSignificantSubdomainCustom('xx.blogspot.co.at', 'public_suffix_list'); -- xx
```

Comment:
Let's add a case with fallback to second level. I suspect it might not work if you have something deep like `aaa.bbb.ccc.ddd.eee.fff`, because you're rotating the `last_3_periods`. (Btw it's also called 'dot' in the comments, I'd use 'dot' everywhere because it is less ambiguous.)

Reply:
>Let's add a case with fallback to second level for long names (more than three dots).

Ok

> I suspect it might not work if you have something deep like aaa.bbb.ccc.ddd.eee.fff, because you're rotating the last_3_periods.

It works correctly, since `last_3_periods` (actually two already) is used only in the fallback. Everything it needs is `prev` for lookup.

```sql
┌─cutToFirstSignificantSubdomainCustom('foo.bar.xx.blogspot.co.at', 'public_suffix_list')─┐
│ xx.blogspot.co.at                                                                       │
└─────────────────────────────────────────────────────────────────────────────────────────┘
```

> (Btw it's also called 'dot' in the comments, I'd use 'dot' everywhere because it is less ambiguous.)

Well, I can convert this, but then it will make the code different with `execute()`, and I'm preferring not to touch unrelated bits in patches but if you insist I can do this surely.

Reply:
> I'm preferring not to touch unrelated bits
Me too, let's keep it as is then.